src/sandstorm/util-test.c++

```diff
@@ -239,5 +239,71 @@ KJ_TEST("SubprocessSet") {
   promiseCat.wait(io.waitScope);
 }
 
+KJ_TEST("raiiOpenAtIfExistsContained") {
```

Comment:
Can you add tests covering the case of following "foo/some-symlink/..", verifying that the `..` goes to the symlink target's parent, not back to `foo`?

I _think_ your approach does this "correctly" (where "correctly" means "matching the traditional kernel behavior even if it's confusing"), since you evaluate each symlink strictly after canonicalizing the preceding path so that it contains no symlinks. But this is a common point of confusion with symlink resolution so worth a test.

Reply:
As it turns out, this *doesn't* work, because we pass the path to `kj::Path::parse` to split it up into segments -- which does the resolution in path space before we start walking the segments ourselves.

Reply:
...I suppose we could split the path ourselves, but it feels a bit sad to introduce extra complexity here when the supervisor already rejects paths with `..` in them anyway. Maybe we should just change the API so it accepts a PathPtr instead of a string, and move the call to `Path::parse` out into the supervisor code.

Reply:
Ok, I went ahead and changed the function to accept a `kj::Path`, which avoids this case (which we don't actually need).